vpr/src/base/read_place.cpp

```diff
@@ -185,6 +186,23 @@ void read_place_body(std::ifstream& placement_file,
         place_ctx.block_locs.resize(cluster_ctx.clb_nlist.blocks().size());
     }
 
+    /*
+     * If placement is being loaded (i.e. reading in a place file),
+     * the grid_blocks data structure has not been initialized yet,
+     * so it is initialized here.
+     */
+    if (is_place_file) {
+        auto& grid_blocks = place_ctx.grid_blocks;
```

Comment:
This should be moved to a helper function that creates grid_blocks and is called in both flows (read place or don't read place).  Could be in place_utils.cpp, or in some other file that deals with grid blocks and related utilities (if such a file exists).